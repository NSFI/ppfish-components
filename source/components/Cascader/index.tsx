import Cascader from './Cascader';
import './style/index.less';

export {
  CascaderOptionType,
  FieldNamesType,
  FilledFieldNamesType,
  CascaderExpandTrigger,
  ShowSearchType,
  CascaderProps,
  CascaderState
} from './interface';

export default Cascader;
