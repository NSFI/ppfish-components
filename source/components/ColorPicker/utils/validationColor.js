export default function validationColor(props, propName, componentName) {
  if (props[propName] && !(/^#[0-9a-fA-F]{3,6}$/).test(props[propName])) {
    return new Error(`${componentName}.props.${propName} Validation failed!`);
  }
}
