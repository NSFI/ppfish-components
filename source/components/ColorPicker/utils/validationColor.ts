export default function validationColor(
  props: object,
  propName: string,
  componentName: string
) {
  if (props[propName] && !/^#[0-9a-fA-F]{3,6}$/.test(props[propName])) {
    return new Error(`${componentName}.props.${propName} Validation failed!`);
  }
}
