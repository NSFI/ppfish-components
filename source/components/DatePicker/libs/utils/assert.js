import { ExtendableError } from './errors'

class ErrorConditionFailed extends ExtendableError {
  constructor(...args) {
    super(args)
  }
}

export function require_condition(condition, msg = 'pre-condition failed') {
  if (!condition) {
    throw new ErrorConditionFailed(msg)
  }
}
