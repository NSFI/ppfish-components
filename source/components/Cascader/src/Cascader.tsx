import * as React from 'react';
import Trigger from 'rc-trigger';
import Menus from './Menus';
import { KeyCode, shallowEqualArrays } from '../../../utils';
import arrayTreeFilter from 'array-tree-filter';
import BUILT_IN_PLACEMENTS from './placements';
import { CascaderOptionType, RcCascaderProps, SelectEvent } from '../interface';
import useControlledState from '../../../hooks/useControlledState';

const defaultFieldNames = Object.freeze({ label: 'label', value: 'value', children: 'children' });

const RcCascader: React.FC<RcCascaderProps> = props => {
  const triggerRef = React.useRef<any>();
  // useImperativeHandle(ref, () => ({
  //   getPopupDOMNode() {
  //     return triggerRef.current.getPopupDOMNode();
  //   },
  // }))

  const { defaultValue, options, onPopupVisibleChange, loadData } = props;
  const [value, setValue] = useControlledState([], { value: props.value, defaultValue });
  const [activeValue, setActiveValue] = React.useState([]);

  React.useEffect(() => {
    if (!loadData) {
      setActiveValue(props.value);
    }
  }, [props]);

  const [popupVisible, setPopupVisible] = useControlledState(false, {
    value: props.popupVisible,
    onChange: onPopupVisibleChange,
  });

  const handlePopupVisibleChange = (popupVisible: boolean) => {
    setPopupVisible(popupVisible);
  };

  const getFieldName = name => {
    return props.fieldNames[name] || defaultFieldNames[name];
  };

  const isKeyDownEvent = (e: SelectEvent): e is React.KeyboardEvent => e.type === 'keydown';
  const isConfirmEventType = (e: SelectEvent) =>
    isKeyDownEvent(e) ? e.keyCode === KeyCode.ENTER : e.type === 'click';

  const getCurrentLevelOptions = () => {
    const { options } = props;
    const result = arrayTreeFilter(
      options,
      (o: CascaderOptionType[], level: number) => o[getFieldName('value')] === activeValue[level],
      { childrenKeyName: getFieldName('children') },
    );
    if (result[result.length - 2]) {
      return result[result.length - 2][getFieldName('children')];
    }
    return [...options].filter(o => !o.disabled);
  };

  const getActiveOptions = (activeValue: string[]) =>
    arrayTreeFilter(
      options || [],
      (o: CascaderOptionType, level: number) => o[getFieldName('value')] === activeValue[level],
      { childrenKeyName: getFieldName('children') },
    );

  const handleChange = (options: CascaderOptionType[], visible: boolean, e: SelectEvent) => {
    const { onChange } = props;

    if (isConfirmEventType(e)) {
      onChange(
        options.map(o => o[getFieldName('value')]),
        options,
      );
      setPopupVisible(visible);
    }
  };

  const handleMenuSelect = (targetOption, menuIndex, e) => {
    const { changeOnSelect, loadData, expandTrigger } = props;
    // Keep focused state for keyboard support
    const triggerNode = triggerRef.current.getRootDomNode();
    if (triggerNode && triggerNode.focus) {
      triggerNode.focus();
    }

    if (!targetOption || targetOption.disabled) {
      return;
    }

    const newActiveValue = activeValue.slice(0, menuIndex + 1);
    newActiveValue[menuIndex] = targetOption[getFieldName('value')];
    // 更新activeValue
    setActiveValue(newActiveValue);

    const activeOptions = getActiveOptions(newActiveValue);

    // loadData
    if (targetOption.isLeaf === false && !targetOption[getFieldName('children')] && loadData) {
      if (changeOnSelect) {
        handleChange(activeOptions, true, e);
      }
      loadData(activeOptions);
      return;
    }

    // e.keyCode === KeyCode.ENTER || e.type === 'click'
    const canSyncValue = isConfirmEventType(e);

    if (!targetOption[getFieldName('children')] || !targetOption[getFieldName('children')].length) {
      handleChange(activeOptions, false, e);
      // set value to activeValue when select leaf option
      canSyncValue && setValue(newActiveValue);
    } else if (changeOnSelect && (e.type === 'click' || e.type === 'keydown')) {
      // add e.type judgement to prevent `onChange` being triggered by mouseEnter
      if (expandTrigger === 'hover') {
        handleChange(activeOptions, false, e);
      } else {
        handleChange(activeOptions, true, e);
      }

      // set value to activeValue on every select
      canSyncValue && setValue(newActiveValue);
    }
  };

  const handleKeyDown = e => {
    const { children } = props;
    // https://github.com/ant-design/ant-design/issues/6717
    // Don't bind keyboard support when children specify the onKeyDown
    if (children && children.props && children.props.onKeyDown) {
      children.props.onKeyDown(e);
      return;
    }

    const newActiveValue = [...activeValue];
    const currentLevel = newActiveValue.length - 1 < 0 ? 0 : newActiveValue.length - 1;
    const currentOptions = getCurrentLevelOptions();
    const currentIndex = currentOptions
      .map(o => o[getFieldName('value')])
      .indexOf(newActiveValue[currentLevel]);
    if (
      e.keyCode !== KeyCode.DOWN &&
      e.keyCode !== KeyCode.UP &&
      e.keyCode !== KeyCode.LEFT &&
      e.keyCode !== KeyCode.RIGHT &&
      e.keyCode !== KeyCode.ENTER &&
      e.keyCode !== KeyCode.BACKSPACE &&
      e.keyCode !== KeyCode.ESC
    ) {
      return;
    }
    // Press any keys above to reopen menu
    if (
      !popupVisible &&
      e.keyCode !== KeyCode.BACKSPACE &&
      e.keyCode !== KeyCode.LEFT &&
      e.keyCode !== KeyCode.RIGHT &&
      e.keyCode !== KeyCode.ESC
    ) {
      setPopupVisible(true);
      return;
    }
    if (e.keyCode === KeyCode.DOWN || e.keyCode === KeyCode.UP) {
      let nextIndex = currentIndex;
      if (nextIndex !== -1) {
        if (e.keyCode === KeyCode.DOWN) {
          nextIndex += 1;
          nextIndex = nextIndex >= currentOptions.length ? 0 : nextIndex;
        } else {
          nextIndex -= 1;
          nextIndex = nextIndex < 0 ? currentOptions.length - 1 : nextIndex;
        }
      } else {
        nextIndex = 0;
      }
      newActiveValue[currentLevel] = currentOptions[nextIndex][getFieldName('value')];
    } else if (e.keyCode === KeyCode.LEFT || e.keyCode === KeyCode.BACKSPACE) {
      newActiveValue.splice(newActiveValue.length - 1, 1);
    } else if (e.keyCode === KeyCode.RIGHT) {
      if (currentOptions[currentIndex] && currentOptions[currentIndex][getFieldName('children')]) {
        newActiveValue.push(
          currentOptions[currentIndex][getFieldName('children')][0][getFieldName('value')],
        );
      }
    } else if (e.keyCode === KeyCode.ESC) {
      props.esc && setPopupVisible(false);
      return;
    }

    if (!newActiveValue || newActiveValue.length === 0) {
      setPopupVisible(false);
    }

    const activeOptions = getActiveOptions(newActiveValue);
    const targetOption = activeOptions[activeOptions.length - 1];
    handleMenuSelect(targetOption, activeOptions.length - 1, e);

    if (props.onKeyDown) {
      props.onKeyDown(e);
    }
  };

  const {
    disabled,
    popupPlacement,
    builtinPlacements,
    transitionName,
    prefixCls,
    popupClassName,
    fieldNames,
    dropdownMenuColumnStyle,
    children,
    expandTrigger,
    expandIcon,
  } = props;

  // render Menus
  const isEmptyOptions = !options || options.length === 0;
  const menus = isEmptyOptions ? (
    <div />
  ) : (
    <Menus
      prefixCls={prefixCls}
      dropdownMenuColumnStyle={dropdownMenuColumnStyle}
      value={value}
      activeValue={activeValue}
      options={options}
      expandTrigger={expandTrigger}
      onSelect={handleMenuSelect}
      visible={popupVisible}
      defaultFieldNames={defaultFieldNames}
      fieldNames={fieldNames}
      expandIcon={expandIcon}
    />
  );

  const emptyMenuClassName = isEmptyOptions ? ` ${prefixCls}-menus-empty` : '';

  return (
    <Trigger
      ref={triggerRef}
      popupPlacement={popupPlacement}
      builtinPlacements={builtinPlacements}
      popupTransitionName={transitionName}
      action={disabled ? [] : ['click']}
      popupVisible={disabled ? false : popupVisible}
      onPopupVisibleChange={handlePopupVisibleChange}
      prefixCls={`${prefixCls}-menus`}
      popupClassName={popupClassName + emptyMenuClassName}
      popup={menus}
    >
      {React.cloneElement(children, {
        onKeyDown: handleKeyDown,
        tabIndex: disabled ? undefined : 0,
      })}
    </Trigger>
  );
};

RcCascader.defaultProps = {
  options: [],
  onChange() {},
  onPopupVisibleChange() {},
  disabled: false,
  transitionName: '',
  prefixCls: 'rc-cascader',
  popupClassName: '',
  popupPlacement: 'bottomLeft',
  builtinPlacements: BUILT_IN_PLACEMENTS,
  expandTrigger: 'click',
  fieldNames: { label: 'label', value: 'value', children: 'children' },
  expandIcon: '',
  esc: true,
};

export default RcCascader;
