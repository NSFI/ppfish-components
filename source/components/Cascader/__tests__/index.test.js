import React from 'react';
import { render, mount } from 'enzyme';
import { KeyCode } from '../../../utils';
import Cascader from '../index';
import { act } from 'react-test-renderer';

const options = [
  {
    value: 'zhejiang',
    label: 'Zhejiang',
    children: [
      {
        value: 'hangzhou',
        label: 'Hangzhou',
        children: [
          {
            value: 'xihu',
            label: 'West Lake',
          },
        ],
      },
    ],
  },
  {
    value: 'jiangsu',
    label: 'Jiangsu',
    children: [
      {
        value: 'nanjing',
        label: 'Nanjing',
        children: [
          {
            value: 'zhonghuamen',
            label: 'Zhong Hua Men',
          },
        ],
      },
    ],
  },
];

function filter(inputValue, path) {
  return path.some(option => option.label.toLowerCase().indexOf(inputValue.toLowerCase()) > -1);
}

describe('Cascader', () => {
  it('popup correctly when panel is hidden', () => {
    const wrapper = mount(<Cascader options={options} />);
    expect(render(wrapper.find('Trigger').instance().getComponent())).toMatchSnapshot();
  });

  it('popup correctly when panel is open', () => {
    const onVisibleChange = jest.fn();
    const wrapper = mount(<Cascader options={options} onVisibleChange={onVisibleChange} />);
    wrapper.find('input').simulate('click');
    expect(render(wrapper.find('Trigger').instance().getComponent())).toMatchSnapshot();
    expect(onVisibleChange).toHaveBeenCalledWith(true);
  });

  it('support controlled mode', () => {
    const wrapper = mount(<Cascader options={options} />);
    wrapper.setProps({
      value: ['zhejiang', 'hangzhou', 'xihu'],
    });
    expect(wrapper.render()).toMatchSnapshot();
  });

  it('popup correctly with defaultValue', () => {
    const wrapper = mount(<Cascader options={options} defaultValue={['zhejiang', 'hangzhou']} />);
    wrapper.find('input').simulate('click');
    expect(render(wrapper.find('Trigger').instance().getComponent())).toMatchSnapshot();
  });

  it('should support popupVisible', () => {
    const wrapper = mount(<Cascader options={options} defaultValue={['zhejiang', 'hangzhou']} />);
    expect(wrapper.find('Trigger').instance().getComponent().props.visible).toBe(false);
    wrapper.setProps({ popupVisible: true });
    expect(wrapper.find('Trigger').instance().getComponent().props.visible).toBe(true);
  });

  it('can be selected', () => {
    const onChange = jest.fn();
    const wrapper = mount(<Cascader options={options} onChange={onChange} />);
    wrapper.find('input').simulate('click');
    let popupWrapper = mount(wrapper.find('Trigger').instance().getComponent());
    popupWrapper
      .find('.fishd-cascader-menu')
      .at(0)
      .find('.fishd-cascader-menu-item')
      .at(0)
      .simulate('click');
    expect(render(wrapper.find('Trigger').instance().getComponent())).toMatchSnapshot();
    popupWrapper = mount(wrapper.find('Trigger').instance().getComponent());
    popupWrapper
      .find('.fishd-cascader-menu')
      .at(1)
      .find('.fishd-cascader-menu-item')
      .at(0)
      .simulate('click');
    expect(render(wrapper.find('Trigger').instance().getComponent())).toMatchSnapshot();
    popupWrapper = mount(wrapper.find('Trigger').instance().getComponent());
    popupWrapper
      .find('.fishd-cascader-menu')
      .at(2)
      .find('.fishd-cascader-menu-item')
      .at(0)
      .simulate('click');
    expect(render(wrapper.find('Trigger').instance().getComponent())).toMatchSnapshot();
    expect(onChange).toHaveBeenCalledWith(['zhejiang', 'hangzhou', 'xihu'], expect.anything());
  });

  it('backspace should work with `Cascader[showSearch]`', () => {
    const wrapper = mount(<Cascader options={options} showSearch />);
    wrapper.find('input').simulate('change', { target: { value: '123' } });
    expect(wrapper.find('input').prop('value')).toBe('123');
    wrapper.find('input').simulate('keydown', { keyCode: KeyCode.BACKSPACE });
    // Simulate onKeyDown will not trigger onChange by default, so the value is still '123'

    expect(wrapper.find('input').prop('value')).toBe('123');
  });

  it('should highlight keyword and filter when search in Cascader', () => {
    const wrapper = mount(<Cascader options={options} showSearch={{ filter }} />);
    wrapper.find('input').simulate('click');
    wrapper.find('input').simulate('change', { target: { value: 'z' } });
    expect(wrapper.find('input').prop('value')).toBe('z');
    const popupWrapper = mount(wrapper.find('Trigger').instance().getComponent());
    expect(popupWrapper).toMatchSnapshot();
  });

  it('should render not found content', () => {
    const wrapper = mount(<Cascader options={options} showSearch={{ filter }} />);
    wrapper.find('input').simulate('click');
    wrapper.find('input').simulate('change', { target: { value: '__notfoundkeyword__' } });
    expect(wrapper.find('input').prop('value')).toBe('__notfoundkeyword__');
    const popupWrapper = mount(wrapper.find('Trigger').instance().getComponent());
    expect(popupWrapper).toMatchSnapshot();
  });

  it('should support to clear selection', () => {
    const wrapper = mount(<Cascader options={options} defaultValue={['zhejiang', 'hangzhou']} />);
    expect(wrapper.find('.fishd-cascader-picker-label').text()).toBe('Zhejiang / Hangzhou');
    wrapper.find('.fishd-cascader-picker-clear').at(0).simulate('click');
    expect(wrapper.find('.fishd-cascader-picker-label').text()).toBe('');
  });

  it('should close popup when clear selection', () => {
    const onVisibleChange = jest.fn();
    const wrapper = mount(
      <Cascader
        options={options}
        popupVisible
        defaultValue={['zhejiang', 'hangzhou']}
        onVisibleChange={onVisibleChange}
      />,
    );
    wrapper.find('.fishd-cascader-picker-clear').at(0).simulate('click');
    expect(onVisibleChange).toHaveBeenCalledWith(false);
  });

  it('should clear search input when clear selection', () => {
    const wrapper = mount(
      <Cascader options={options} defaultValue={['zhejiang', 'hangzhou']} showSearch />,
    );
    wrapper.find('input').simulate('click');
    wrapper.find('input').simulate('change', { target: { value: 'xxx' } });
    expect(wrapper.find('input').prop('value')).toBe('xxx');
    wrapper.find('.fishd-cascader-picker-clear').at(0).simulate('click');
    expect(wrapper.find('input').prop('value')).toBe('');
  });

  it('should not trigger visible change when click search input', () => {
    const onVisibleChange = jest.fn();
    const wrapper = mount(
      <Cascader options={options} showSearch onVisibleChange={onVisibleChange} />,
    );

    wrapper.find('input').simulate('focus');
    expect(onVisibleChange).toHaveBeenCalledTimes(0);

    wrapper.find('input').simulate('click');
    expect(onVisibleChange).toHaveBeenCalledTimes(1);

    wrapper.find('input').simulate('click');
    expect(onVisibleChange).toHaveBeenCalledTimes(1);

    wrapper.find('input').simulate('blur');
    const mousedownEvent = new MouseEvent('mousedown', {
      bubbles: true,
      cancelable: true,
      view: window,
    });
    document.body.dispatchEvent(mousedownEvent);
    expect(onVisibleChange).toHaveBeenCalledTimes(2);
  });

  it('should change filtered item when options are changed', () => {
    const wrapper = mount(<Cascader options={options} showSearch={{ filter }} />);
    wrapper.find('input').simulate('click');
    wrapper.find('input').simulate('change', { target: { value: 'a' } });

    expect(wrapper.find('.fishd-cascader-menu-item').length).toBe(2);

    wrapper.setProps({ options: [options[0]] });
    wrapper.update();
    expect(wrapper.find('.fishd-cascader-menu-item').length).toBe(1);
  });
});
