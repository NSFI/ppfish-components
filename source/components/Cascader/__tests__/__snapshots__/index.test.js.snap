// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cascader can be selected 1`] = `
<div>
  <div
    class="fishd-cascader-menus"
    style="opacity:0"
  >
    <div>
      <ul
        class="fishd-cascader-menu"
      >
        <li
          class="fishd-cascader-menu-item fishd-cascader-menu-item-expand fishd-cascader-menu-item-active"
          title="Zhejiang"
        >
          Zhejiang
          <span
            class="fishd-cascader-menu-item-expand-icon"
          />
        </li>
        <li
          class="fishd-cascader-menu-item fishd-cascader-menu-item-expand"
          title="Jiangsu"
        >
          Jiangsu
          <span
            class="fishd-cascader-menu-item-expand-icon"
          />
        </li>
      </ul>
      <ul
        class="fishd-cascader-menu"
      >
        <li
          class="fishd-cascader-menu-item fishd-cascader-menu-item-expand"
          title="Hangzhou"
        >
          Hangzhou
          <span
            class="fishd-cascader-menu-item-expand-icon"
          />
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`Cascader can be selected 2`] = `
<div>
  <div
    class="fishd-cascader-menus"
    style="opacity:0"
  >
    <div>
      <ul
        class="fishd-cascader-menu"
      >
        <li
          class="fishd-cascader-menu-item fishd-cascader-menu-item-expand fishd-cascader-menu-item-active"
          title="Zhejiang"
        >
          Zhejiang
          <span
            class="fishd-cascader-menu-item-expand-icon"
          />
        </li>
        <li
          class="fishd-cascader-menu-item fishd-cascader-menu-item-expand"
          title="Jiangsu"
        >
          Jiangsu
          <span
            class="fishd-cascader-menu-item-expand-icon"
          />
        </li>
      </ul>
      <ul
        class="fishd-cascader-menu"
      >
        <li
          class="fishd-cascader-menu-item fishd-cascader-menu-item-expand fishd-cascader-menu-item-active"
          title="Hangzhou"
        >
          Hangzhou
          <span
            class="fishd-cascader-menu-item-expand-icon"
          />
        </li>
      </ul>
      <ul
        class="fishd-cascader-menu"
      >
        <li
          class="fishd-cascader-menu-item"
          title="West Lake"
        >
          West Lake
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`Cascader can be selected 3`] = `<div />`;

exports[`Cascader popup correctly when panel is hidden 1`] = `<div />`;

exports[`Cascader popup correctly when panel is open 1`] = `
<div>
  <div
    class="fishd-cascader-menus"
    style="opacity:0"
  >
    <div>
      <ul
        class="fishd-cascader-menu"
      >
        <li
          class="fishd-cascader-menu-item fishd-cascader-menu-item-expand"
          title="Zhejiang"
        >
          Zhejiang
          <span
            class="fishd-cascader-menu-item-expand-icon"
          />
        </li>
        <li
          class="fishd-cascader-menu-item fishd-cascader-menu-item-expand"
          title="Jiangsu"
        >
          Jiangsu
          <span
            class="fishd-cascader-menu-item-expand-icon"
          />
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`Cascader popup correctly with defaultValue 1`] = `
<div>
  <div
    class="fishd-cascader-menus"
    style="opacity:0"
  >
    <div>
      <ul
        class="fishd-cascader-menu"
      >
        <li
          class="fishd-cascader-menu-item fishd-cascader-menu-item-expand fishd-cascader-menu-item-active"
          title="Zhejiang"
        >
          Zhejiang
          <span
            class="fishd-cascader-menu-item-expand-icon"
          />
        </li>
        <li
          class="fishd-cascader-menu-item fishd-cascader-menu-item-expand"
          title="Jiangsu"
        >
          Jiangsu
          <span
            class="fishd-cascader-menu-item-expand-icon"
          />
        </li>
      </ul>
      <ul
        class="fishd-cascader-menu"
      >
        <li
          class="fishd-cascader-menu-item fishd-cascader-menu-item-expand fishd-cascader-menu-item-active"
          title="Hangzhou"
        >
          Hangzhou
          <span
            class="fishd-cascader-menu-item-expand-icon"
          />
        </li>
      </ul>
      <ul
        class="fishd-cascader-menu"
      >
        <li
          class="fishd-cascader-menu-item"
          title="West Lake"
        >
          West Lake
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`Cascader should highlight keyword and filter when search in Cascader 1`] = `
<Popup
  align={
    Object {
      "offset": Array [
        0,
        8,
      ],
      "overflow": Object {
        "adjustX": 1,
        "adjustY": 1,
      },
      "points": Array [
        "tl",
        "bl",
      ],
    }
  }
  className=""
  destroyPopupOnHide={false}
  getClassNameFromAlign={[Function]}
  getRootDomNode={[Function]}
  mask={false}
  onAlign={[Function]}
  onMouseDown={[Function]}
  onTouchStart={[Function]}
  prefixCls="fishd-cascader-menus"
  style={Object {}}
  transitionName="slide-up"
  visible={true}
>
  <div>
    <Mask
      align={
        Object {
          "offset": Array [
            0,
            8,
          ],
          "overflow": Object {
            "adjustX": 1,
            "adjustY": 1,
          },
          "points": Array [
            "tl",
            "bl",
          ],
        }
      }
      className=""
      destroyPopupOnHide={false}
      getClassNameFromAlign={[Function]}
      getRootDomNode={[Function]}
      mask={false}
      onAlign={[Function]}
      onMouseDown={[Function]}
      onTouchStart={[Function]}
      prefixCls="fishd-cascader-menus"
      style={Object {}}
      transitionName="slide-up"
      visible={true}
    />
    <PopupInner
      align={
        Object {
          "offset": Array [
            0,
            8,
          ],
          "overflow": Object {
            "adjustX": 1,
            "adjustY": 1,
          },
          "points": Array [
            "tl",
            "bl",
          ],
        }
      }
      className=""
      destroyPopupOnHide={false}
      getClassNameFromAlign={[Function]}
      getRootDomNode={[Function]}
      mask={false}
      onAlign={[Function]}
      onMouseDown={[Function]}
      onTouchStart={[Function]}
      prefixCls="fishd-cascader-menus"
      style={Object {}}
      transitionName="slide-up"
      visible={true}
    >
      <CSSMotion
        leavedClassName="fishd-cascader-menus-hidden"
        motionName="slide-up"
        onAppearEnd={[Function]}
        onAppearPrepare={[Function]}
        onEnterEnd={[Function]}
        onEnterPrepare={[Function]}
        onLeaveEnd={[Function]}
        removeOnLeave={false}
        visible={true}
      >
        <DomWrapper>
          <Align
            align={
              Object {
                "offset": Array [
                  0,
                  8,
                ],
                "overflow": Object {
                  "adjustX": 1,
                  "adjustY": 1,
                },
                "points": Array [
                  "tl",
                  "bl",
                ],
              }
            }
            disabled={true}
            key="popup"
            monitorWindowResize={true}
            onAlign={[Function]}
            target={[Function]}
          >
            <div
              className="fishd-cascader-menus"
              onMouseDownCapture={[Function]}
              onTouchStartCapture={[Function]}
              style={
                Object {
                  "opacity": 0,
                  "pointerEvents": undefined,
                  "zIndex": undefined,
                }
              }
            >
              <Menus
                activeValue={Array []}
                defaultFieldNames={
                  Object {
                    "children": "children",
                    "label": "label",
                    "value": "value",
                  }
                }
                dropdownMenuColumnStyle={
                  Object {
                    "width": 0,
                  }
                }
                expandIcon=""
                expandTrigger="click"
                fieldNames={
                  Object {
                    "children": "children",
                    "label": "label",
                    "value": "value",
                  }
                }
                onSelect={[Function]}
                options={
                  Array [
                    Object {
                      "__IS_FILTERED_OPTION": true,
                      "disabled": false,
                      "label": Array [
                        "Jiangsu",
                        Array [
                          " / ",
                          "Nanjing",
                        ],
                        Array [
                          " / ",
                          "Zhong Hua Men",
                        ],
                      ],
                      "path": Array [
                        Object {
                          "children": Array [
                            Object {
                              "children": Array [
                                Object {
                                  "label": "Zhong Hua Men",
                                  "value": "zhonghuamen",
                                },
                              ],
                              "label": "Nanjing",
                              "value": "nanjing",
                            },
                          ],
                          "label": "Jiangsu",
                          "value": "jiangsu",
                        },
                        Object {
                          "children": Array [
                            Object {
                              "label": "Zhong Hua Men",
                              "value": "zhonghuamen",
                            },
                          ],
                          "label": "Nanjing",
                          "value": "nanjing",
                        },
                        Object {
                          "label": "Zhong Hua Men",
                          "value": "zhonghuamen",
                        },
                      ],
                      "value": Array [
                        "jiangsu",
                        "nanjing",
                        "zhonghuamen",
                      ],
                    },
                    Object {
                      "__IS_FILTERED_OPTION": true,
                      "disabled": false,
                      "label": Array [
                        "Zhejiang",
                        Array [
                          " / ",
                          Array [
                            "Hang",
                            Array [
                              <span
                                className="fishd-cascader-menu-item-keyword"
                              >
                                z
                              </span>,
                              "hou",
                            ],
                          ],
                        ],
                        Array [
                          " / ",
                          "West Lake",
                        ],
                      ],
                      "path": Array [
                        Object {
                          "children": Array [
                            Object {
                              "children": Array [
                                Object {
                                  "label": "West Lake",
                                  "value": "xihu",
                                },
                              ],
                              "label": "Hangzhou",
                              "value": "hangzhou",
                            },
                          ],
                          "label": "Zhejiang",
                          "value": "zhejiang",
                        },
                        Object {
                          "children": Array [
                            Object {
                              "label": "West Lake",
                              "value": "xihu",
                            },
                          ],
                          "label": "Hangzhou",
                          "value": "hangzhou",
                        },
                        Object {
                          "label": "West Lake",
                          "value": "xihu",
                        },
                      ],
                      "value": Array [
                        "zhejiang",
                        "hangzhou",
                        "xihu",
                      ],
                    },
                  ]
                }
                prefixCls="fishd-cascader"
                value={Array []}
                visible={true}
              >
                <div>
                  <ul
                    className="fishd-cascader-menu"
                    key="0"
                    style={
                      Object {
                        "width": 0,
                      }
                    }
                  >
                    <li
                      className="fishd-cascader-menu-item"
                      key="jiangsu,nanjing,zhonghuamen"
                      onClick={[Function]}
                      title=""
                    >
                      Jiangsu
                       / 
                      Nanjing
                       / 
                      Zhong Hua Men
                    </li>
                    <li
                      className="fishd-cascader-menu-item"
                      key="zhejiang,hangzhou,xihu"
                      onClick={[Function]}
                      title=""
                    >
                      Zhejiang
                       / 
                      Hang
                      <span
                        className="fishd-cascader-menu-item-keyword"
                        key="seperator"
                      >
                        z
                      </span>
                      hou
                       / 
                      West Lake
                    </li>
                  </ul>
                </div>
              </Menus>
            </div>
          </Align>
        </DomWrapper>
      </CSSMotion>
    </PopupInner>
  </div>
</Popup>
`;

exports[`Cascader should render not found content 1`] = `
<Popup
  align={
    Object {
      "offset": Array [
        0,
        8,
      ],
      "overflow": Object {
        "adjustX": 1,
        "adjustY": 1,
      },
      "points": Array [
        "tl",
        "bl",
      ],
    }
  }
  className=""
  destroyPopupOnHide={false}
  getClassNameFromAlign={[Function]}
  getRootDomNode={[Function]}
  mask={false}
  onAlign={[Function]}
  onMouseDown={[Function]}
  onTouchStart={[Function]}
  prefixCls="fishd-cascader-menus"
  style={Object {}}
  transitionName="slide-up"
  visible={true}
>
  <div>
    <Mask
      align={
        Object {
          "offset": Array [
            0,
            8,
          ],
          "overflow": Object {
            "adjustX": 1,
            "adjustY": 1,
          },
          "points": Array [
            "tl",
            "bl",
          ],
        }
      }
      className=""
      destroyPopupOnHide={false}
      getClassNameFromAlign={[Function]}
      getRootDomNode={[Function]}
      mask={false}
      onAlign={[Function]}
      onMouseDown={[Function]}
      onTouchStart={[Function]}
      prefixCls="fishd-cascader-menus"
      style={Object {}}
      transitionName="slide-up"
      visible={true}
    />
    <PopupInner
      align={
        Object {
          "offset": Array [
            0,
            8,
          ],
          "overflow": Object {
            "adjustX": 1,
            "adjustY": 1,
          },
          "points": Array [
            "tl",
            "bl",
          ],
        }
      }
      className=""
      destroyPopupOnHide={false}
      getClassNameFromAlign={[Function]}
      getRootDomNode={[Function]}
      mask={false}
      onAlign={[Function]}
      onMouseDown={[Function]}
      onTouchStart={[Function]}
      prefixCls="fishd-cascader-menus"
      style={Object {}}
      transitionName="slide-up"
      visible={true}
    >
      <CSSMotion
        leavedClassName="fishd-cascader-menus-hidden"
        motionName="slide-up"
        onAppearEnd={[Function]}
        onAppearPrepare={[Function]}
        onEnterEnd={[Function]}
        onEnterPrepare={[Function]}
        onLeaveEnd={[Function]}
        removeOnLeave={false}
        visible={true}
      >
        <DomWrapper>
          <Align
            align={
              Object {
                "offset": Array [
                  0,
                  8,
                ],
                "overflow": Object {
                  "adjustX": 1,
                  "adjustY": 1,
                },
                "points": Array [
                  "tl",
                  "bl",
                ],
              }
            }
            disabled={true}
            key="popup"
            monitorWindowResize={true}
            onAlign={[Function]}
            target={[Function]}
          >
            <div
              className="fishd-cascader-menus"
              onMouseDownCapture={[Function]}
              onTouchStartCapture={[Function]}
              style={
                Object {
                  "opacity": 0,
                  "pointerEvents": undefined,
                  "zIndex": undefined,
                }
              }
            >
              <Menus
                activeValue={Array []}
                defaultFieldNames={
                  Object {
                    "children": "children",
                    "label": "label",
                    "value": "value",
                  }
                }
                dropdownMenuColumnStyle={
                  Object {
                    "height": "auto",
                    "width": 0,
                  }
                }
                expandIcon=""
                expandTrigger="click"
                fieldNames={
                  Object {
                    "children": "children",
                    "label": "label",
                    "value": "value",
                  }
                }
                onSelect={[Function]}
                options={
                  Array [
                    Object {
                      "disabled": true,
                      "label": "无匹配结果",
                      "value": "FISHD_CASCADER_NOT_FOUND",
                    },
                  ]
                }
                prefixCls="fishd-cascader"
                value={Array []}
                visible={true}
              >
                <div>
                  <ul
                    className="fishd-cascader-menu"
                    key="0"
                    style={
                      Object {
                        "height": "auto",
                        "width": 0,
                      }
                    }
                  >
                    <li
                      className="fishd-cascader-menu-item fishd-cascader-menu-item-disabled"
                      key="FISHD_CASCADER_NOT_FOUND"
                      onClick={[Function]}
                      title="无匹配结果"
                    >
                      无匹配结果
                    </li>
                  </ul>
                </div>
              </Menus>
            </div>
          </Align>
        </DomWrapper>
      </CSSMotion>
    </PopupInner>
  </div>
</Popup>
`;

exports[`Cascader support controlled mode 1`] = `
<span
  class="fishd-cascader-picker"
  tabindex="0"
>
  <span
    class="fishd-cascader-picker-label"
    title="Zhejiang / Hangzhou / West Lake"
  >
    Zhejiang / Hangzhou / West Lake
  </span>
  <input
    autocomplete="off"
    class="fishd-input fishd-cascader-input "
    readonly=""
    type="text"
    value=""
  />
  <i
    class="fishdicon fishdicon-close-circle-fill fishd-cascader-picker-clear"
  />
  <i
    class="fishdicon fishdicon-down-fill fishd-cascader-picker-arrow"
  />
</span>
`;
